### PR TITLE
Use for range loops instead of manual for loops

### PR DESCRIPTION
**Pull request by tmthrgd**
This replaces the use of manual `for i := 0; i < len(...); i++` loops with simpler and more idiomatic `for range` loops.

`Msg.CopyTo` had a slight reorganisation of it's logic which leaves it much simpler: 5bab01b0a655730908f4523a5b2510d80a515167.

### CODE REVIEW

File: clientconfig.go [hunk 1]
```diff
@@ -68,14 +68,10 @@ func ClientConfigFromReader(resolvconf io.Reader) (*ClientConfig, error) {
 			}
 
 		case "search": // set search path to given servers
-			c.Search = make([]string, len(f)-1)
-			for i := 0; i < len(c.Search); i++ {
-				c.Search[i] = f[i+1]
-			}
+			c.Search = append([]string(nil), f[1:]...)
```
Comment (miekg) [inline clientconfig.go]: `string(nil)`? can't use `[]string{}` ?

File: dnssec.go [hunk 2]
```diff
@@ -556,9 +556,9 @@ func (k *DNSKEY) publicKeyRSA() *rsa.PublicKey {
 	pubkey := new(rsa.PublicKey)
 
 	var expo uint64
-	for i := 0; i < int(explen); i++ {
+	for _, v := range keybuf[keyoff:modoff] {
```
Comment (miekg) [inline dnssec.go]: found that `explen` give some extra documentation here, add comment that this loops is executed `explen` times?

Comment (tmthrgd) [reply dnssec.go]: @miekg Can do. I’ll document it tomorrow when I’m back at my computer.

File: clientconfig.go (hunk 1 above)
Comment (tmthrgd) [reply clientconfig.go]: Idiomatic go is to use `nil` here. It needs the type of the slice, but not the actual slice. It’s straight out of the wiki: https://github.com/golang/go/wiki/SliceTricks#copy.

File: dnssec.go (hunk 2 above)
Comment (tmthrgd) [reply dnssec.go]: @miekg How is that comment?

**Pull request merged**